Update environment-variables.md (#7134)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/environment-variables.md

- `DBT_ENV` &mdash; This key is reserved for the dbt Cloud application and will always resolve to 'prod'. For deployment runs only.
- `DBT_CLOUD_ENVIRONMENT_NAME` &mdash; The name of the dbt Cloud environment in which `dbt` is running. 

> - `DBT_CLOUD_ENVIRONMENT_TYPE` &mdash; The type of dbt Cloud environment in which `dbt` is running. The valid values are `dev`, `staging`, or `prod`. The value will be empty for General deployment environments, so use a default like `{{ env_var('DBT_CLOUD_ENVIRONMENT_TYPE', '') }}`.

Reviewer: ```suggestion
- `DBT_CLOUD_ENVIRONMENT_TYPE` &mdash; The type of dbt Cloud environment where `dbt` is running. The valid values are `dev`, `staging`, or `prod`. The value is empty for [General deployment environments](/docs/dbt-cloud-environments#types-of-environments), so use a default like `{{ env_var('DBT_CLOUD_ENVIRONMENT_TYPE', '') }}`.
```
